### PR TITLE
Errors in notification config should not crash the server

### LINKED ISSUE #6870: Minio goes into crash loop if mqtt credentials fail
I have minio running in kubernetes v1.12
The k8s manifest can be seen here: https://gist.github.com/cbluth/442bea8b6b9e3955bdca973db54360ee#file-minio-deployment-yaml

## Expected Behavior

I expected the minio server to run without crashing, even if it cannot connect to mqtt.
I would expect an error in the log, and the application to continue working like "normal".

## Current Behavior

It goes into crash loop.

## Possible Solution

log the error, and then ignore mqtt events.

## Steps to Reproduce (for bugs)

1. setup minio, here is a sample manifest: https://gist.github.com/cbluth/442bea8b6b9e3955bdca973db54360ee#file-minio-deployment-yaml
2. configure events for mqtt as explained here: https://docs.minio.io/docs/minio-bucket-notification-guide.html#MQTT
3. miss-spell the mqtt credentials in previous step.
4. minio crashes with this error:
```
cbluth@dl7490:~$ kubectl --context=dev -n storage logs minio-7b5bbcc968-94nc4


API: SYSTEM()
Time: 13:04:09 UTC 11/27/2018
Error: Not Authorized
       1: cmd/config-current.go:691:cmd.getNotificationTargets()
       2: cmd/notification.go:462:cmd.NewNotificationSys()
       3: cmd/server-main.go:380:cmd.serverMain()

API: SYSTEM(bucket=cbluth)
Time: 13:04:09 UTC 11/27/2018
Error: ARN 'arn:minio:sqs::1:mqtt' not found
       1: cmd/notification.go:578:cmd.readNotificationConfig()
       2: cmd/notification.go:315:cmd.(*NotificationSys).refresh()
       3: cmd/notification.go:347:cmd.(*NotificationSys).Init()
       4: cmd/server-main.go:383:cmd.serverMain()
ERROR Unable to initialize notification system: ARN 'arn:minio:sqs::1:mqtt' not found.
cbluth@dl7490:~$
```

## Context

I want minio events on my mqtt topic.

## Your Environment

* Version used (`minio version`): `minio/minio:latest`
* Environment name and version (e.g. nginx 1.9.1): k8s v1.12
* Server type and version:
* Operating System and version (`uname -a`): ubuntu 16.0 LTS
* Link to your project:

### PR DESCRIPTION
**Pull request by Praveenrajmani**
Fixes #6870

## Description
Minio server was crashing for errors in setConfig
Please refer #6870 for more info

## Motivation and Context
minio should fallback on such notification errors and continue serving

## Regression
No

## How Has This Been Tested?

1 - Start the minio server with bucket notification enabled
2 - Stop the notification server
3 - Try to restart the server by exiting and spawning again (or) `./mc admin service restart myminio`
4 - The server will emit logs but wont crash

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/server-main.go [hunk 1]
```diff
@@ -380,9 +380,8 @@ func serverMain(ctx *cli.Context) {
 	globalNotificationSys = NewNotificationSys(globalServerConfig, globalEndpoints)
 
 	// Initialize notification system.
-	if err = globalNotificationSys.Init(newObject); err != nil {
-		logger.Fatal(err, "Unable to initialize notification system")
-	}
+	// Ignoring the errors
+	_ = globalNotificationSys.Init(newObject)
```
Comment (kannappanr) [inline cmd/server-main.go]: The error should be logged here.

Comment (Praveenrajmani) [reply cmd/server-main.go]: We already log the notification errors before 

image

Should we still add `Unable to initialize notification config` ?

Comment (harshavardhana) [reply cmd/server-main.go]: We perhaps have to still to correlate the context of the other errors.

Comment (Praveenrajmani) [reply cmd/server-main.go]: :+1:

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 6881-e216461/gopath/src/github.com/minio
+ git clone --quiet https://github.com/minio/minio.git 6881-e216461/gopath/src/github.com/minio/minio
++ cd 6881-e216461/gopath/src/github.com/minio/minio
++ git remote add minio https://github.com/minio/minio.git
++ cd 6881-e216461/gopath/src/github.com/minio/minio
++ git fetch --quiet minio pull/6881/head:pr6881
++ cd 6881-e216461/gopath/src/github.com/minio/minio
++ git checkout --quiet pr6881
++ minikube ip
+ minikube_ip=192.168.39.129
++ mkdir -p certs
++ cd certs
++ ./cert -host 192.168.39.129
2018/12/03 02:46:22 wrote public.crt
2018/12/03 02:46:22 wrote private.key
+ GOPATH=/mnt/nvme0n1p1/mint-auto/mint-auto/6881-e216461/gopath
+ make -C 6881-e216461/gopath/src/github.com/minio/minio --quiet
# github.com/minio/minio/cmd
cmd/server-main.go:385:15: cannot use ctx (type *cli.Context) as type context.Context in argument to logger.LogIf:
	*cli.Context does not implement context.Context (missing Deadline method)
make: *** [build] Error 2

```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**